src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -222,60 +239,167 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
+        })
+        .map(ar -> ar.isEmpty() ? null : parseErmTitle(ar.getJsonObject(0)));
+  }
+
+  Future<Tuple> ermTitleLookup(RoutingContext ctx, UUID id) {
+    String uri = "/erm/resource/" + id;
+    return getRequest(ctx, uri)
+        .send()
+        .compose(res -> {
+          if (res.statusCode() != 200) {
+            return Future.failedFuture(uri + " returned " + res.statusCode());
+          }
+          return Future.succeededFuture(res.bodyAsJsonObject());
+        })
+        .map(EusageReportsApi::parseErmTitle);
+  }
+
+  Future<List<UUID>> ermPackageContentLookup(RoutingContext ctx, UUID id) {
+    // example: /erm/packages/dfb61870-1252-4ece-8f75-db02faf4ab82/content
+    String uri = "/erm/packages/" + id + "/content";
+    return getRequest(ctx, uri)
         .send()
         .compose(res -> {
           if (res.statusCode() != 200) {
             return Future.failedFuture(uri + " returned " + res.statusCode());
           }
           JsonArray ar = res.bodyAsJsonArray();
-          return Future.succeededFuture(ar);
+          List<UUID> list = new ArrayList<>();
+          for (int i = 0; i < ar.size(); i++) {
+            UUID kbTitleId = UUID.fromString(ar.getJsonObject(i).getJsonObject("pti")
+                .getJsonObject("titleInstance").getString("id"));
+            list.add(kbTitleId);
+          }
+          return Future.succeededFuture(list);
         });
-    return future.compose(ar -> {
-      if (ar.isEmpty()) {
-        return Future.succeededFuture(null);
-      }
-      JsonObject resource = ar.getJsonObject(0);
-      UUID titleId = UUID.fromString(resource.getString("id"));
-      return Future.succeededFuture(Tuple.of(titleId, resource.getString("name")));
-    });
   }
 
+
   Future<UUID> upsertTeEntry(TenantPgPool pool, SqlConnection con, RoutingContext ctx,
                              String counterReportTitle, String printIssn, String onlineIssn) {
-    return con.preparedQuery("SELECT id FROM " + titleEntriesTable(pool)
-        + " WHERE counterReportTitle = $1")
-        .execute(Tuple.of(counterReportTitle))
+    final String match = onlineIssn != null ? "online:" + onlineIssn : "print:" + printIssn;
+    return con.preparedQuery("SELECT id,counterReportTitle FROM " + titleEntriesTable(pool)
+        + " WHERE matchCriteria = $1")
+        .execute(Tuple.of(match))
         .compose(res1 -> {
           if (res1.iterator().hasNext()) {
-            return Future.succeededFuture(res1.iterator().next().getUUID(0));
+            Row row = res1.iterator().next();
+            UUID id = row.getUUID(0);
+            if (row.getString(1) != null) {  // matched with counter reports already?
+              return Future.succeededFuture(id);
+            }
+            return con.preparedQuery("UPDATE " + titleEntriesTable(pool)
+                + " SET"
+                + " counterReportTitle = $2,"
+                + " printISSN = $4,"
+                + " onlineISSN = $5"
```

Comment:
printISSN = $3
onlineISSN = $4

Reply:
yep. This part is not covered by test, so it should be.

Reply:
Now covered by tests and in doing so I found one more grave error. (now fixed)